### PR TITLE
Repository structure and documentation cleanup

### PR DESCRIPTION
**Pull request by gferon**
Following the merge of `flatdata-rs`, I felt that we had to move documentation in their appropriate subfolder, merge back some from the Rust impl. and reword a few things.

TODO:
- [x] Check that all tests run as expected
- [x] Finalize the "catchy slogan"
- [x] Make sure the `docs/` directory is up-to-date / non-redundant before merging

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,31 +16,19 @@ Flatdata helps creating efficient datasets:
...
-make
-make test  # optional
-```
-
 ## Using `flatdata`
 
-*Note: Until the release APIs are expected to be undergoing (potentially breaking) changes. Binary data layout is stable, though.*
+### Creating a schema
 
 Define a flatdata archive, let's say `locations.flatdata`:
```
Comment (boxdot) [inline README.md]: I guess the file name is not important anymore.

File: README.md [hunk 2]
```diff
@@ -55,79 +43,37 @@ namespace loc {
...
 
-## Library Layout
+The following languages are supported:
 
-Library is organized as follows:
+  * First-class citizen implementations:
+    * **[C++](./flatdata-cpp)** - used extensively, tested excessively normally receives features first
+    * **[Rust](./flatdata-rs)** - the newest addition to the family
+  * Read-only implementations:
+    * **[Python](./flatdata-py)** - used mostly for debugging purposes
```
Comment (boxdot) [inline README.md]: I would say python is rather for `inspecting the data`. Not if it is more to the point. Feel free not to change.

File: README.md [hunk 3]
```diff
@@ -140,3 +86,9 @@ distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
+
+### Contribution
+
+Unless you explicitly state otherwise, any contribution intentionally submitted
+for inclusion in this document by you, as defined in the Apache-2.0 license,
+shall be dual licensed as above, without any additional terms or conditions.
```
Comment (boxdot) [inline README.md]: -dual

File: flatdata-py/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+## Basic usage
+
+Once you have [created a flatdata schema file](../README.md#creating-a-schema), you can generate a Python module:
+
+```sh
+./generator/app.py --gen python --schema locations.flatdata --output-file locations.py
+```
+
+TODO: Add basic python usages
```
Comment (boxdot) [inline flatdata-py/README.md]: Just a note: There is also inspect flatdata file in tools. I would actually take a look at this file and consider to provide instructions instead how to use flatdata from jupyter.

File: flatdata-rs/README.md [hunk 5]
```diff
@@ -1,20 +1,6 @@
-# flatdata-rs [![travis status]][travis] [![latest version]][crates.io] [![docs]][docs.rs]
+# flatdata-rs [![latest version]][crates.io] [![docs]][docs.rs]
```
Comment (boxdot) [inline flatdata-rs/README.md]: Note: Need to adjust the path in the toml definition of the crate.

Comment (gferon) [reply flatdata-rs/README.md]: what do you mean?

File: flatdata-py/README.md (hunk 4 above)
Comment (gferon) [reply flatdata-py/README.md]: Sounds like a good idea. In general, we need to check why `flatdata-py` depends on the internal modules of the `generator` - with this proper split of directories, you either need to run those scripts from the root, or manipulate `PYTHONPATH`, which I don't like very much (it's never very obvious).

File: flatdata-rs/README.md (hunk 5 above)
Comment (boxdot) [reply flatdata-rs/README.md]: I mean that `Cargo.toml` contains a path to the repository, which has now changed. (Just commented on this file due to the lack of context.)

**Comment by gferon** [comment]
There are still 2 markdown files under `docs/` - what do you guys think is the best way to integrate it with the `README.md`?

File: README.md [hunk 6]
```diff
@@ -1,48 +1,36 @@
...
-* [Why Flatdata](#why-flatdata)
-* [Building Flatdata](#building-flatdata)
-* [Using Flatdata](#using-flatdata)
-* [Library Layout](#library-layout)
-* [License](#license)
+Flatdata is a library providing data structures for convenient creation, storage and access of packed memory-mappable structures with minimal overhead.
+
+With `flatdata`, the user defines a schema of the data format using a very simple schema language that supports plain structs, vectors and multivectors. The schema is then used to generate builders and readers for serialization and deserialization of the data to an archive of files on disk.
+
+The data is serialized in a portable way which allows zero-overhead random access to it by using memory mapped storage: the operating system facilities for loading, caching and paging of the data, and most important, accessing it as if it were in memory.
```
Comment (VeaaC) [inline README.md]: in a portable while while allowing zero-overhead

Comment (VeaaC) [reply README.md]: operating system facilities... -> verb is missing?

File: flatdata-cpp/README.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+make
+make test # optional
+```
+
+## Usage & basic example
+
+Once you have [created a flatdata schema file](../README.md#creating-a-schema), you can generate a C++ header:
+
+```shell
+./generator/app.py --gen cpp --schema locations.flatdata --output-file locations.hpp
```
Comment (VeaaC) [inline flatdata-cpp/README.md]: locations.flatdata schema inline here (since the rest of the example uses it)?

File: flatdata-py/README.md (hunk 4 above)
Comment (VeaaC) [reply flatdata-py/README.md]: flatdata-py is 90% generator and just a little bit on top if I am not mistaken (it basically uses the AST the generator has internally)

File: flatdata-py/tests/__init__.py [hunk 8]
```diff
@@ -0,0 +1,3 @@
+# to be able to import the generator
+import sys, os
+sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), "../../"))
```
Comment (VeaaC) [inline flatdata-py/tests/__init__.py]: missing newline?

File: flatdata-rs/Cargo.toml [hunk 9]
```diff
@@ -3,8 +3,8 @@ name = "flatdata"
 version = "0.2.1"
 authors = ["boxdot <d@zerovolt.org>"]
 license = "Apache-2.0"
-description = "Rust implementation of heremaps/flatdata"
-repository = "https://github.com/boxdot/flatdata-rs"
+description = "Rust implementation of flatdata"
+repository = "https://github.com/heremaps/flatdata"
```
Comment (VeaaC) [inline flatdata-rs/Cargo.toml]: Does the sub-path need to be mentioned here?

File: tests/CMakeLists.txt [hunk 10]
```diff
@@ -1,21 +0,0 @@
-include(flatdata/PythonRequirements)
```
Comment (VeaaC) [inline tests/CMakeLists.txt]: Where did this actually move?

File: flatdata-rs/Cargo.toml (hunk 9 above)
Comment (gferon) [reply flatdata-rs/Cargo.toml]: This is designed to perform a `git clone`, so the subpath is not part of this field.

File: flatdata-py/README.md (hunk 4 above)
Comment (gferon) [reply flatdata-py/README.md]: then do you think it makes sense to merge back the generator inside of flatdata-py and have a symlink at the root, or a launcher called `generator` or `generator.py`?

File: tests/CMakeLists.txt (hunk 10 above)
Comment (gferon) [reply tests/CMakeLists.txt]: inside of `flatdata-py`

**Pull request merged**